source/distributed/tria.cc

```diff
@@ -3124,92 +3124,14 @@ namespace parallel
-                                                        periodic.face_idx[0], cf,
-                                                        periodic.orientation[0],
-                                                        periodic.orientation[1],
-                                                        periodic.orientation[2]);
-                periodic_child.cell[0] = periodic.cell[0]->child(child_index_0);
-                periodic_child.face_idx[0] = periodic.face_idx[0];
-
-                // the second face is in standard orientation in terms of the
-                // periodic face pair
-                const unsigned int child_index_1 =
-                  GeometryInfo<dim>::child_cell_on_face(periodic.cell[1]->refinement_case(),
-                                                        periodic.face_idx[1], cf);
-
-                periodic_child.cell[1] = periodic.cell[1]->child(child_index_1);
-                periodic_child.face_idx[1] = periodic.face_idx[1];
-
-                // recursive call into children
-                identify_periodic_vertices_recursively (periodic_child,
-                                                        topological_vertex_numbering);
-              }
-          }
-
-        for (unsigned int v=0; v<GeometryInfo<dim-1>::vertices_per_cell; ++v)
-          {
-            // take possible non-standard orientation of face on cell[0] into
-            // account
-            const unsigned int vface0 =
-              GeometryInfo<dim>::standard_to_real_face_vertex(v,periodic.orientation[0],
-                                                              periodic.orientation[1],
-                                                              periodic.orientation[2]);
-            const unsigned int vi0 = topological_vertex_numbering[periodic.cell[0]->face(periodic.face_idx[0])->vertex_index(vface0)];
-            const unsigned int vi1 = topological_vertex_numbering[periodic.cell[1]->face(periodic.face_idx[1])->vertex_index(v)];
-            const unsigned int min_index = std::min(vi0, vi1);
-            topological_vertex_numbering[periodic.cell[0]->face(periodic.face_idx[0])->vertex_index(vface0)]
-              = topological_vertex_numbering[periodic.cell[1]->face(periodic.face_idx[1])->vertex_index(v)]
-                = min_index;
-          }
-      }
-
-
-
       // ensures the 2:1 mesh balance for periodic boundary conditions in the
       // artificial cell layer (the active cells are taken care of by p4est)
       template <int dim, int spacedim>
       bool enforce_mesh_balance_over_periodic_boundaries
       (Triangulation<dim,spacedim> &tria,
-       const std::vector<GridTools::PeriodicFacePair<typename dealii::Triangulation<dim,spacedim>::cell_iterator> > periodic_face_pairs_level_0)
+       const std::vector<GridTools::PeriodicFacePair<typename dealii::Triangulation<dim,spacedim>::cell_iterator> > &periodic_face_pairs_level_0)
```

Comment:
Do you still use periodic_face_pairs_level0 in this function? I can't seem to find it. I guess you only need periodic_face_map here (for which I don't quite understand how you can simply call 'tria.periodic_face_map' below? `enforce_mesh_balance_over_periodic_boundaries` is a separate function, and periodic_face_map is not public?)